models/issue.go

```diff
@@ -1109,10 +1103,19 @@ func newIssue(e *xorm.Session, doer *User, opts NewIssueOptions) (err error) {
 	}
 
 	// Milestone and assignee validation should happen before insert actual object.
-	if _, err = e.Insert(opts.Issue); err != nil {
+	if _, err := e.SetExpr("`index`", "coalesce(MAX(`index`),0)+1").
```

Comment:
Is this really supported in all db types?

Reply:
yes it should

Reply:
Check my comment about DB support here: https://github.com/go-gitea/gitea/pull/7898#issuecomment-523685648

As far as the syntax and the functionality, they are supported. About the atomicity, in theory it's there, but It's very difficult to test. There some WIP about that here #7950 and here #7959.

Reply:
To validate there is an integration `TestAPICreateIssue` that validate the creation across all db types.